src/llmq/signing_shares.cpp

```diff
@@ -1479,15 +1479,15 @@ void CSigSharesManager::WorkThreadMain()
 
 void CSigSharesManager::AsyncSign(const CQuorumCPtr& quorum, const uint256& id, const uint256& msgHash)
 {
-    LOCK(cs);
+    LOCK(cs_pendingSigns);
     pendingSigns.emplace_back(quorum, id, msgHash);
 }
 
 void CSigSharesManager::SignPendingSigShares()
 {
     std::vector<PendingSignatureData> v;
     {
-        LOCK(cs);
+        LOCK(cs_pendingSigns);
```

Comment:
I just noticed, that there can be potential bug, because all standard library objects that have been moved from are placed in a "valid but unspecified state", meaning the object's class invariants hold (so functions without preconditions, such as the assignment operator, can be safely used on the object after it was moved from). See example:
 
```
std::vector<int> v = {2, 3, 3};
std::vector<int> u = std::move(v); // the value of v is unspecified
v.clear(); // we are good now
```

Instead, let's have swap better!

```
WITH_LOCK(cs_pendingSigns, v.swap(pendingSigns));
```

Reply:
addressed here: https://github.com/dashpay/dash/pull/6433